engine/src/main/java/com/datatorrent/stram/plan/physical/PhysicalPlan.java

```diff
@@ -895,8 +895,10 @@ private void redoPartitions(PMapping currentMapping, String note)
         addedPartitions.add(newPartition);
       } else {
         // check whether mapping was changed
+        int currentPartitionsSize = mainPC.currentPartitions.size();
         for (DefaultPartition<Operator> pi : mainPC.currentPartitions) {
-          if (pi == newPartition && pi.isModified()) {
+          if (pi == newPartition && (pi.isModified() ||
```

Comment:
@vrozov The problem occurs when definPartitions() returns the list of partitions (size > 1)  which consists of current partition of size 1.  Unifier will come into the picture in newPartitions  which is not be there in initial launch. Here, port mapping of current partition which is in mainPC.newPartitions has to be changed.